lib/Net/SAML2/Protocol/Assertion.pm

```diff
@@ -161,6 +161,7 @@ sub new_from_xml {
     }
 
     my $self = $class->new(
+        id             => $xpath->findvalue('//saml:Assertion/@ID'),
```

Comment:
Could we add a test for this in t/03-assertions.t, if we have that, it can be merged imo.

Reply:
Will do, thought it was going to be a bigger issue so I did a separate test that checked the id of all from_xml uses.  I can add that and update  t/03-assertions.t